mathics/core/expression.py

```diff
@@ -636,14 +636,12 @@ def evaluate(
         self,
         evaluation: Evaluation,
     ) -> typing.Type["BaseElement"]:
-        """Apply transformation rules and expression evaluation to `evaluation` via
-        `rewrite_apply_eval_step()` until it tells us to stop or we hit some limit.
-
-        Note that this is a recusive process and
-        `rewrite_apply_eval_step()` may call us recursively.
-
-        Limits are either an evaluation iteration count or a timeout value.
+        """
+        Apply transformation rules and expression evaluation to ``evaluation`` via
+        ``rewrite_apply_eval_step()`` until that method tells us to stop,
+        or unti we hit an $IterationLimit or TimeConstrained limit.
```

Comment:
I think this sasy exactly the same thing but is shorter and while also specific, e.g. mention of  `$IterationLimit` and `TimeConstrained`